tests/basis-ecp/input.dat

```diff
@@ -0,0 +1,56 @@
+memory 1000 mb
+
+set df_fitting_condition 1e-12
+
+molecule {
+-1 1
+ I
+}
+
+set basis def2-tzvppd
+
+ene, wfn = energy('scf', return_wfn=True)
+compare_integers(25, wfn.molecule().Z(0), "Zeff")  #TEST
+compare_integers(13, wfn.nalpha(), "nalpha")  #TEST
+compare_integers(28, wfn.basisset().n_ecp_core(), "n_ecp_core")  #TEST
+compare_values(-296.7441153128044107, ene, 5, "ecp basis")  #TEST
+
+clean()
+
+set basis 3-21g
+
+ene, wfn = energy('scf', return_wfn=True)
+compare_integers(53, wfn.molecule().Z(0), "Zeff")  #TEST
+compare_integers(27, wfn.nalpha(), "nalpha")  #TEST
+compare_integers(0, wfn.basisset().n_ecp_core(), "n_ecp_core")  #TEST
+compare_values(-6896.649171106428, ene, 5, "all e- basis after ecp")  #TEST
+
+clean()
+
+set basis def2-tzvppd
+set basis_guess true
+
+# will run after #1484 merged
+#ene, wfn = energy('scf', return_wfn=True)
+#compare_integers(25, wfn.molecule().Z(0), "Zeff")  #TEST
+#compare_integers(13, wfn.nalpha(), "nalpha")  #TEST
+#compare_integers(28, wfn.basisset().n_ecp_core(), "n_ecp_core")  #TEST
+#compare_values(-296.7441153128044107, ene, 5, "cast-up with ecp after all e- basis")  #TEST
+
+
+clean()
+
+set basis_guess 3-21g
+
+try:
+    energy('scf')
+except ValidationError as e:
+    compare_integers(True, 'ECP electrons will be a disaster' in str(e), 'caught bad cast-up basis')  #TEST
```

Comment:
This is a very clever approach.  We should add an `assert_throws` method to our comparison helpers that will do this (in another PR).  Something like
``` python
def assert_throws(fn, args, exception_type=Exception, expected_message=""):
    try:
        fn(*args)
    except exception_type as e:
        if expected_message in str(e):
            return True
        else:
            print("Exception did not contain expected message")
            return False
    except Exception as e:
        print("Expected exception did not get raised")
        return False
```
but without the errors that are surely present in this code that I just made up

Reply:
oh yeah, if we do this it'd need an extra statement after the `try/except` to handle the case where it doesn't throw at all, of course

Reply:
I entirely agree with that strategy in general. But I think it'd be better to do it in `pytest`. That already has `with pytest.raises(exceptiontype):` and `with pytest.warns(warningtype:` constructs, and I turn all our `compare_*` functions into T/F for use with `assert` https://github.com/psi4/psi4/blob/master/tests/pytest/utils.py .